Polkalokr Web3 Grant Application

anischohan opened the pull request:
# Grant Application Checklist

- [x] The [application-template.md](https://github.com/w3f/Open-Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( "project_name.md") and updated.
- [x] A BTC or Ethereum address for the payment of the milestones is provided inside the application.  
- [x] The software of the project will be released under the Apache license version 2.0 as specified in the [terms and conditions](https://gist.github.com/Noc2/75bc58e8ce9b5d419ff883b0cf2b8c19).
- [x] The total funding amount of the project is below USD $30k for initial grant applications and $100k for follow-up grants at the time of submission.
- [x] The initial PR contains only one commit (squash if needed before submitting your PR).
- [x] The grant will only be announced once we successfully [delivered](https://github.com/w3f/Grant-Milestone-Delivery) the first milestone.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Open-Grants-Program?pullRequest=229) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Open-Grants-Program?pullRequest=229) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Open-Grants-Program?pullRequest=229) it.</sub>

Noc2 reviewed (changes requested):
Thanks for the application. Which smart contract language are you planning to use? Are you aware of the [Vesting Module](https://github.com/paritytech/substrate/tree/master/frame/vesting)?

anischohan commented:
Thanks for reviewing our application so far Noc2. We've responded to your queries and updated the application.md file. If you require any further clarifications feel free to get in touch. Look forward to hearing from you.

Noc2 reviewed (changes requested):
Sorry for the late reply and thanks for the update. It would be nice if you could already clearly say which smart contract language you are going to use, because we generally like to support new smart contract languages like ink! via our grant programs. Feel free to remove the deployment on rococo from the requirements, since this is something we can not ensure. Would it be possible to also deliver some UI as part of the grant, because it’s currently rather expensive? Potentially it also makes sense to integrate the PSP process into this application: https://github.com/w3f/PSPs to work on some kind standardisation. Additionally, I recommend switching to a payment in DAI. This way we don’t need to constantly update the price.

anischohan commented:
> Sorry for the late reply and thanks for the update. It would be nice if you could already clearly say which smart contract language you are going to use, because we generally like to support new smart contract languages like ink! via our grant programs. Feel free to remove the deployment on rococo from the requirements, since this is something we can not ensure. Would it be possible to also deliver some UI as part of the grant, because it’s currently rather expensive? Potentially it also makes sense to integrate the PSP process into this application: https://github.com/w3f/PSPs to work on some kind standardisation. Additionally, I recommend switching to a payment in DAI. This way we don’t need to constantly update the price.

Thank you for reviewing our updates @Noc2. We have updated the proposal as requested. We now clearly state ink!. We have added UI, and PSP and removed rococo testnet from deliverables. We have also updated with DAI price.

Noc2 commented on applications/polkalokr.md:
```diff
@@ -0,0 +1,200 @@
+| 5c. ||GetTokensAndLock(addresses[], currencyToLock, amount)|
+| 5d. ||ReleaseToken(addresses[], currencyToUnlock, amount)|
+| 5e. ||UpdateRecipientAddress(contractAddress, addresses[]) |  
+| 6. | User Interface | We will design, and develop a simple interface that demonstrates smart contract functionality |
```
Could you clarify which smart contract functionality the UI supports (or will it support all of the above features)?  Also do you know which programming language you will use and do you potentially already have some initial designs?

Noc2 commented on applications/polkalokr.md:
```diff
@@ -0,0 +1,200 @@
+| 5d. ||ReleaseToken(addresses[], currencyToUnlock, amount)|
+| 5e. ||UpdateRecipientAddress(contractAddress, addresses[]) |  
+| 6. | User Interface | We will design, and develop a simple interface that demonstrates smart contract functionality |
+| 7. | PSP | We will write up a proposal for the PSP iniative. |  
```
Could you clarify the psp initiative a little bit better?

Noc2 reviewed (changes requested):
Thanks for the update and the nice call today. Just a few follow-up questions, see below

Noc2 commented:
I’m closing this PR due to inactivity. Let me know if you want to reopen it.
